src/sentry/static/sentry/app/views/organizationIntegrations/integrationRow.tsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import Link from 'app/components/links/link';
+import {PanelItem} from 'app/components/panels';
+import PluginIcon from 'app/plugins/components/pluginIcon';
+import space from 'app/styles/space';
+import {Organization, SentryApp} from 'app/types';
+import {t} from 'app/locale';
+
+import IntegrationStatus from './integrationStatus';
+
+type Props = {
+  organization: Organization;
+  type: 'plugin' | 'firstParty' | 'sentryApp';
+  slug: string;
+  displayName: string;
+  status: 'Installed' | 'Not Installed' | 'Pending';
+  publishStatus: 'unpublished' | 'published' | 'internal';
```

Comment:
I'm not sure how difficult it would be, but I think the `|` types should be formalizes into their own types, maybe in `app/types`.

If right now, we wanted to add a new `status` or something, it would be really difficult to make sure every location using this type was updated, since there's no source of truth (i.e. things like `getAppInstallStatus` in IntegrationListDirectory.tsx don't reference this type, they just happen to match.

Reply:
Totally might be out of scope of this PR btw, don't change it if it's a big thing

Reply:
@leeandher I am working on making a type for this in one of my PRs so don't worry about that :)